File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>

void givechange(float);
int main(void)

{
    float c;
    do
    {
        printf("How much change is owed (in dollars):");
        c=get_float();
    }
    while (c<0);
    
    givechange(c);
}

    void givechange(float c)
    {
        int cents=c*100;
        int a=0;
        while (cents>24)
        {
            a++;
            cents=cents-25;
        }
        int b=0;
        while (cents>9)
        {
            b++;
            cents=cents-10;
        }
        int d=0;
        while (cents>4)
        {
            d++;
            cents=cents-5;
        }
        int e=0;
        while (cents>0)
        {
            e++;
            cents=cents-1;
        }
        printf("No. of 25c: %i\n", a);
        printf("No. of 10c: %i\n", b);
        printf("No. of 5c: %i\n", d);
        printf("No. of 1c: %i\n", e);
        printf("%i\n", a+b+d+e);
    }
    

File: pset6/greedy.py
import cs50

def main():
    while True:
        print("How much change is owed (in dollars):", end="")
        c=cs50.get_float()
        if c>=0:
            break
        
    givechange(c)
    
def givechange(c):
    cents=c*100
    a=0
    while True:
        cents-=25
        a=a+1
        if cents<25:
            break
    b=0
    while True:
        cents-=10
        b=b+1
        if cents<10:
            break
    d=0
    while True:
        cents-=5
        d=d+1
        if cents<5:
            break
    e=0
    while True:
        cents-=1
        e=e+1
        if cents<1:
            break
    print("No. of 25c: {}".format(a))
    print("No. of 10c: {}".format(b))
    print("No. of 5c: {}".format(d))
    print("No. of 1c: {}".format(e))
    
if __name__ == "__main__":
    main()
    
    